### PR TITLE
oo/214128 deploy application storageaccount

### PR DESCRIPTION
**Pull request by osagiestar**

### CODE REVIEW

**Review by aadev1 — APPROVED** [review]

File: infra/storage-account/application-storage-account.parameters.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "storageAccount": {
+            "value": {
+                "name": "#{{ infraResourceNamePrefix }}#{{ nc_resource_storageaccount }}#{{ nc_instance_regionid }}01",
+                "privateEndpointName": "#{{ infraResourceNamePrefix }}#{{ nc_resource_privateendpoint }}#{{ nc_instance_regionid }}05",
+                "skuName": "Standard_ZRS"
```
[the remark this answers is not in the document]
Comment (osagiestar) [reply infra/storage-account/application-storage-account.parameters.json:9]: Thankk you @ajaysoni-cap . I can parameterise it if required. @DaRoszko gave ZRS as the value to be applied during one of our meetings.

File: infra/storage-account/application-storage-account.bicep [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  'StorageV2'
+  'BlobStorage'
+  'FileStorage'
+  'BlockBlobStorage'
+])
+@description('Optional. Type of Storage Account to create for the storage account.')
+param kind string = 'StorageV2'
+
+@description('Optional. Sets the custom domain name assigned to the storage account. Name is the CNAME source.')
+param customDomainName string = ''
```
Comment (kbabigumira) [inline infra/storage-account/application-storage-account.bicep:24]: The custom domain parameters are not being used, they should be removed. I think we need to keep the wrapper bicep modules with minimal parameters. if need arises in future, we can add more parameters.

File: infra/storage-account/application-storage-account.bicep [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+var storageAccountTags = {
+  Name: storageAccount.name
+  Purpose: 'Storage Account'
+  Tier: 'Shared'
+}
+
+var storageAccountPrivateEndpointTags = {
+  Name: storageAccount.privateEndpointName
+  Purpose: 'App Configuration private endpoint'
```
Comment (kbabigumira) [inline infra/storage-account/application-storage-account.bicep:59]: As discussed, can you update the Purpose tag for this bicep module and the application key vault too

File: infra/storage-account/application-storage-account.parameters.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (DaRoszko) [reply infra/storage-account/application-storage-account.parameters.json:9]: zrs is fine

File: infra/storage-account/application-storage-account.bicep (hunk 2 above)
Comment (osagiestar) [reply infra/storage-account/application-storage-account.bicep]: @kbabigumira I have implemented it now.

File: infra/storage-account/application-storage-account.bicep (hunk 3 above)
Comment (osagiestar) [reply infra/storage-account/application-storage-account.bicep]: Sorted

File: infra/storage-account/application-storage-account.parameters.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (osagiestar) [reply infra/storage-account/application-storage-account.parameters.json:9]: Thankl you @DaRoszko

[the remark this answers is not in the document]
Comment (osagiestar) [reply infra/storage-account/application-storage-account.parameters.json:9]: Thank you @ajaysoni-cap for your input

File: infra/storage-account/application-storage-account.bicep (hunk 3 above)
Comment (osagiestar) [reply infra/storage-account/application-storage-account.bicep]: Thank you @kbabigumira for your input

File: infra/storage-account/application-storage-account.parameters.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (kbabigumira) [reply infra/storage-account/application-storage-account.parameters.json:9]: Discussed with @ajaysoni-cap  @DaRoszko @ppotturi  Let's use ZRS for this one.

File: infra/storage-account/application-storage-account.bicep [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+@description('Optional. Indicates whether indirect CName validation is enabled. This should only be set on updates.')
+param customDomainUseSubDomainName bool = false
+
+@description('Optional. Allows you to specify the type of endpoint. Set this to AzureDNSZone to create a large number of accounts in a single subscription, which creates accounts in an Azure DNS Zone and the endpoint URL will have an alphanumeric DNS Zone identifier.')
+@allowed([
+  ''
+  'AzureDnsZone'
+  'Standard'
+])
+param dnsEndpointType string = ''
```
Comment (kbabigumira) [inline infra/storage-account/application-storage-account.bicep:32]: Let's use the minimal number of parameters, don't think we need customDomainUseSubDomainName and dnsEndpointType also. If we need them in future, we can add more parameters to the wrapper bicep module.

Comment (osagiestar) [reply infra/storage-account/application-storage-account.bicep]: @kbabigumira  I have made the changes👍

**Review by DaRoszko — APPROVED** [review]

**Review by ajaysoni-cap — APPROVED** [review]

**Pull request merged**